### PR TITLE
[Merged by Bors] - feat: two BigO lemmas for complex functions resticted to the real line

### PR DESCRIPTION
**Pull request by MichaelStollBayreuth**
This second PR in the series discussed [here](https://leanprover.zulipchat.com/#narrow/channel/144837-PR-reviews/topic/Prerequisites.20for.20PNT.20and.20Dirichlet's.20Thm/near/482005581) adds to lemmas converting Big-O statements for complex functions to their restriction to the real line.

From [EulerProducts](https://github.com/MichaelStollBayreuth/EulerProducts). 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Complex/RealDeriv.lean [hunk 1]
```diff
@@ -170,3 +170,21 @@ theorem conformalAt_iff_differentiableAt_or_differentiableAt_comp_conj {f : ℂ
...
+
+open Topology
+
+lemma isBigO_comp_ofReal {f g : ℂ → ℂ} {x : ℝ} (h : f =O[𝓝 (x : ℂ)] g) :
+    (fun y : ℝ ↦ f y) =O[𝓝 x] (fun y : ℝ ↦ g y) :=
+  h.comp_tendsto <| continuous_ofReal.tendsto x
+
+lemma isBigO_comp_ofReal_nhds_ne {f g : ℂ → ℂ} {x : ℝ} (h : f =O[𝓝[≠] (x : ℂ)] g) :
+    (fun y : ℝ ↦ f y) =O[𝓝[≠] x] (fun y : ℝ ↦ g y) :=
+  h.comp_tendsto <| ((hasDerivAt_id (x : ℂ)).comp_ofReal).tendsto_punctured_nhds one_ne_zero
```
Comment (j-loreaux) [inline Mathlib/Analysis/Complex/RealDeriv.lean:186]: There is no need for derivatives here, and these don't belong in this file as a result.
```suggestion
  h.comp_tendsto <| continuous_ofReal.continuousWithinAt.tendsto_nhdsWithin fun _ _ ↦ by simp_all
```

Comment (MichaelStollBayreuth) [reply Mathlib/Analysis/Complex/RealDeriv.lean]: OK; moved to `Analysis.Complex.Asymptotics`.

File: Mathlib/Analysis/Complex/Asymptotics.lean [hunk 2]
```diff
@@ -43,4 +43,14 @@ theorem isTheta_ofReal_left {f : α → ℝ} {g : α → E} : (f · : α → ℂ
 theorem isTheta_ofReal_right {f : α → E} {g : α → ℝ} : f =Θ[l] (g · : α → ℂ) ↔ f =Θ[l] g :=
   (isTheta_ofReal g l).isTheta_congr_right
 
+open Topology
+
+lemma isBigO_comp_ofReal {f g : ℂ → ℂ} {x : ℝ} (h : f =O[𝓝 (x : ℂ)] g) :
```
Comment (j-loreaux) [inline Mathlib/Analysis/Complex/Asymptotics.lean:48]:
```suggestion
lemma isBigO_comp_ofReal_nhds {f g : ℂ → ℂ} {x : ℝ} (h : f =O[𝓝 (x : ℂ)] g) :
```

**Review by j-loreaux — COMMENTED** [review]
Thanks! Please wait for CI before merging.

bors d+

**Pull request closed without merging**